DotNET: Deprecated signal causes build failure

### Tested versions

- Reproducible in latest master (f1c0b5b8542afb750236b8d35d533737b883de29)

### System information

Windows 10 - DotNET SDK v8.0.300

### Issue description

See the build failure in https://github.com/godotengine/godot/actions/runs/12914187787/job/36013250581?pr=90049

My PR #90049 deprecates a signal on SceneTree which seems to cause a warning (which is treated as error).

```
B:\Godot\modules\mono\glue\GodotSharp\GodotSharp\Generated\GodotObjects\SceneTree.cs(856,11): warning CS0618: 'SceneTree.NodeC
onfigurationWarningChangedEventHandler' is obsolete: 'This signal is no longer available. Call 'update_configuration_info' on  
a 'Node' or 'Resource' to notify the editor of configuration info changes.' [B:\Godot\modules\mono\glue\GodotSharp\GodotSharp\
GodotSharp.csproj]
```

Silencing this warning should be the way to fix this I assume? Generated code snippet for reference:

```cs
    /// <summary>
    /// Represents the method that handles the <see cref="Godot.SceneTree.NodeConfigurationWarningChanged"/> event of a <see cref="Godot.SceneTree"/> class.
    /// </summary>
    [Obsolete("This signal is no longer available. Call 'update_configuration_info' on a 'Node' or 'Resource' to notify the editor of configuration info changes.")]
    public delegate void NodeConfigurationWarningChangedEventHandler(Node node);

    private static void NodeConfigurationWarningChangedTrampoline(object delegateObj, NativeVariantPtrArgs args, out godot_variant ret)
    {
        Callable.ThrowIfArgCountMismatch(args, 1);
        ((NodeConfigurationWarningChangedEventHandler)delegateObj)(VariantUtils.ConvertTo<Node>(args[0]));
        // ^^^^ This usage of the EventHandler type emits the warning!
        ret = default;
    }

    /// <summary>
    /// <para>Emitted when the <c>node</c>'s <see cref="Godot.Node.UpdateConfigurationWarnings()"/> is called. Only emitted in the editor.</para>
    /// </summary>
    [Obsolete("This signal is no longer available. Call 'update_configuration_info' on a 'Node' or 'Resource' to notify the editor of configuration info changes.")]
    public unsafe event NodeConfigurationWarningChangedEventHandler NodeConfigurationWarningChanged
    {
        add => Connect(SignalName.NodeConfigurationWarningChanged, Callable.CreateWithUnsafeTrampoline(value, &NodeConfigurationWarningChangedTrampoline));
        remove => Disconnect(SignalName.NodeConfigurationWarningChanged, Callable.CreateWithUnsafeTrampoline(value, &NodeConfigurationWarningChangedTrampoline));
    }

    protected void EmitSignalNodeConfigurationWarningChanged(Node node)
    {
        EmitSignal(SignalName.NodeConfigurationWarningChanged, node);
    }
```

### Steps to reproduce

- Checkout #90049
- Follow the steps on here: https://docs.godotengine.org/en/latest/contributing/development/compiling/compiling_with_dotnet.html
  - Compile with mono module enabled
  - Generate glue
  - Run build_assemblies.py

### Minimal reproduction project (MRP)

N/A

--- a/modules/mono/editor/bindings_generator.cpp
+++ b/modules/mono/editor/bindings_generator.cpp
@@ -3216,13 +3216,12 @@ Error BindingsGenerator::_generate_cs_signal(const BindingsGenerator::TypeInterf
 			p_output.append(" class.\n");
 			p_output.append(INDENT1 "/// </summary>");
 
+			// Generate delegate
 			if (p_isignal.is_deprecated) {
 				p_output.append(MEMBER_BEGIN "[Obsolete(\"");
 				p_output.append(bbcode_to_text(p_isignal.deprecation_message, &p_itype));
 				p_output.append("\")]");
 			}
-
-			// Generate delegate
 			p_output.append(MEMBER_BEGIN "public delegate void ");
 			p_output.append(delegate_name);
 			p_output.append("(");
@@ -3230,6 +3229,11 @@ Error BindingsGenerator::_generate_cs_signal(const BindingsGenerator::TypeInterf
 			p_output.append(");\n");
 
 			// Generate Callable trampoline for the delegate
+			if (p_isignal.is_deprecated) {
+				p_output.append(MEMBER_BEGIN "[Obsolete(\"");
+				p_output.append(bbcode_to_text(p_isignal.deprecation_message, &p_itype));
+				p_output.append("\")]");
+			}
 			p_output << MEMBER_BEGIN "private static void " << p_isignal.proxy_name << "Trampoline"
 					 << "(object delegateObj, NativeVariantPtrArgs args, out godot_variant ret)\n"
 					 << INDENT1 "{\n"
@@ -3279,17 +3283,16 @@ Error BindingsGenerator::_generate_cs_signal(const BindingsGenerator::TypeInterf
 			}
 		}
 
-		if (p_isignal.is_deprecated) {
-			p_output.append(MEMBER_BEGIN "[Obsolete(\"");
-			p_output.append(bbcode_to_text(p_isignal.deprecation_message, &p_itype));
-			p_output.append("\")]");
-		}
-
 		// TODO:
 		// Could we assume the StringName instance of signal name will never be freed (it's stored in ClassDB) before the managed world is unloaded?
 		// If so, we could store the pointer we get from `data_unique_pointer()` instead of allocating StringName here.
 
 		// Generate event
+		if (p_isignal.is_deprecated) {
+			p_output.append(MEMBER_BEGIN "[Obsolete(\"");
+			p_output.append(bbcode_to_text(p_isignal.deprecation_message, &p_itype));
+			p_output.append("\")]");
+		}
 		p_output.append(MEMBER_BEGIN "public ");
 
 		if (p_itype.is_singleton) {
@@ -3339,6 +3342,11 @@ Error BindingsGenerator::_generate_cs_signal(const BindingsGenerator::TypeInterf
 
 		// Generate EmitSignal{EventName} method to raise the event.
 		if (!p_itype.is_singleton) {
+			if (p_isignal.is_deprecated) {
+				p_output.append(MEMBER_BEGIN "[Obsolete(\"");
+				p_output.append(bbcode_to_text(p_isignal.deprecation_message, &p_itype));
+				p_output.append("\")]");
+			}
 			p_output.append(MEMBER_BEGIN "protected void ");
 			p_output << "EmitSignal" << p_isignal.proxy_name;
 			if (is_parameterless) {

